src/Type/AcceptsResult.php

```diff
@@ -118,14 +126,30 @@ public static function maxMin(self ...$operands): self
 		}
 
 		$result = TrinaryLogic::maxMin(...array_map(static fn (self $result) => $result->result, $operands));
+
+		return new self($result, self::mergeReasons($operands));
+	}
+
+	public function negate(): self
+	{
+		return new self($this->result->negate(), $this->reasons);
+	}
```

Comment:
I dunno if it makes sens to have a negate method (Should the reasons be lost ?)

I created one since I had multiple Trinary::negate calls to replace.